posthog/src/main/java/com/posthog/java/PostHog.java

```diff
@@ -0,0 +1,155 @@
+package com.posthog.java;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Collections;
+import org.json.JSONObject;
+import org.json.JSONException;
+import java.net.URI;
+
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okhttp3.Call;
+import okhttp3.MultipartBody;
+
+public class PostHog {
+    private String apiKey;
+    private String host;
+
+    public static class Builder {
+        // required
+        private final String apiKey;
+
+        // optional
+        // private String host = "https://app.posthog.com";
+        private String host = "localhost";
+
+        public Builder(String apiKey) {
+            this.apiKey = apiKey;
+        }
+
+        public Builder host(String value) {
+            host = value;
+            return this;
+        }
+
+        public PostHog build() {
+            return new PostHog(this);
+        }
+    }
+
+    private PostHog(Builder builder) {
+        apiKey = builder.apiKey;
+        host = builder.host;
+    }
+
+    private void send(String distinctId, String event, HashMap<String, Object> properties, String propName) {
```

Comment:
let's not have `propName` and `properties` 

`properties`, `$set`, and `$set_once` are not mutually exclusive. I'd add a HashMap for each _or_ you can also put `$set` and `$set_once` _inside_ `properties`. They can be both top-level or nested.